source/flow.hpp

```diff
@@ -12,7 +12,7 @@ class ISender;
 class IFlow {
 public:
     virtual void start(Time time) = 0;
-    virtual bool try_to_generate(Size packet_size) = 0;
+    virtual std::optional<Time> try_to_generate() = 0;
```

Comment:
Rollback this interface change. it optimises time of simulator work but occures many unneseccary changes. Moreover, not is is not essential optimisation.
```suggestion
    virtual Time try_to_generate() = 0;
```